File: 2418-sort-the-people/2418-sort-the-people.cpp
class Solution {
public:
    vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
        vector<pair<int,string>>vec;
        for(int i=0;i<names.size();i++){
            vec.push_back({heights[i],names[i]});
        }
        
        sort(vec.begin(), vec.end(), greater<>());
        vector<string> result;
        for(auto [i,j]:vec){
            result.push_back(j);
        }
        return result;
    }
};

File: 2418-sort-the-people/2418-sort-the-people.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        int n = names.length;
        
        TreeMap<Integer, String> map = new TreeMap<>();
        
        for(int i = 0; i < n; i++){
            map.put(heights[i], names[i]);
        }
        
        int idx = n - 1;
        
        String[] ans = new String[n];
        
        for(int i : map.keySet()){
            ans[idx] = map.get(i);
            idx--;
        }
        
        return ans;
    }
}